File: Compose/YABA/app/src/main/java/dev/subfly/yaba/core/model/annotation/AnnotationQuoteSnapshot.kt
package dev.subfly.yaba.core.model.annotation

import androidx.compose.runtime.Stable
import kotlinx.serialization.Serializable

/**
 * Best-effort text capture for a selection or saved annotation.
 * Used for UI preview in creation sheet and overview.
 */
@Serializable
@Stable
data class AnnotationQuoteSnapshot(
    val selectedText: String,
    val prefixText: String? = null,
    val suffixText: String? = null,
) {
    val displayText: String get() = selectedText.trim().ifBlank { "" }

    companion object {
        fun fromSelectedText(text: String): AnnotationQuoteSnapshot =
            AnnotationQuoteSnapshot(selectedText = text.trim())
    }
}

File: Darwin/YABACore/Model/Annotation/AnnotationQuoteSnapshot.swift
//
//  AnnotationQuoteSnapshot.swift
//  YABACore
//
//  Parity with Compose `AnnotationQuoteSnapshot`.
//

import Foundation

public struct AnnotationQuoteSnapshot: Sendable, Codable, Equatable {
    public var selectedText: String
    public var prefixText: String?
    public var suffixText: String?

    public init(selectedText: String, prefixText: String? = nil, suffixText: String? = nil) {
        self.selectedText = selectedText
        self.prefixText = prefixText
        self.suffixText = suffixText
    }

    public var displayText: String {
        selectedText.trimmingCharacters(in: .whitespacesAndNewlines)
    }

    public static func fromSelectedText(_ text: String) -> AnnotationQuoteSnapshot {
        AnnotationQuoteSnapshot(
            selectedText: text.trimmingCharacters(in: .whitespacesAndNewlines)
        )
    }
}
